doc/manual/src/glossary.md

```diff
@@ -85,12 +85,18 @@
 
     [store path]: #gloss-store-path
 
+  - [file system object]{#gloss-store-object}\
+    The Nix data model for representing file system data.
+
+    See [File System Object](@docroot@/architecture/file-system-object.md) for details.
+
+    [file system object]: #gloss-file-system-object
+
   - [store object]{#gloss-store-object}\
-    A file that is an immediate child of the Nix store directory. These
-    can be regular files, but also entire directory trees. Store objects
-    can be sources (objects copied from outside of the store),
-    derivation outputs (objects produced by running a build task), or
-    derivations (files describing a build task).
+
+    A store object is a [file system object] with [reference]s to other store objects.
```

Comment:
```suggestion
    A store object combines a [file system object] with [reference]s to other store objects.
```

```suggestion
    A store object is formed by a [file system object] and [reference]s to other store objects.
```

.. and we also store other metadata, like a creation timestamp, deriver (should be derivers), maybe more.

Reply:
Are those things covered by a signature? I never really considered them part of the core data model. It's good to think about!

Reply:
The fingerprint consists of the file system object and the references but none of the metadata that I mentioned.
So they don't have to be part of the _core_ data model, but I don't think we should imply that the metadata doesn't exist either. The glossary shouldn't be exclusively about an idealized data model.

Reply:
```suggestion
    A store object consists of a [file system object], [reference]s to other store objects, and other metadata.
```

Reply:
Sounds good. For now it is all fairly tight in-line in the glossary, but we can better separate it out once it gets its own page.